node/pkg/chain/common/common_test.go

```diff
@@ -0,0 +1,106 @@
+package common
+
+import (
+	"testing"
+)
+
+func TestParseMethodSignature(t *testing.T) {
+	tests := []struct {
+		name           string
+		input          string
+		expectedName   string
+		expectedInput  string
+		expectedOutput string
+		expectedError  error
+	}{
+		{
+			name:           "Test case 1",
+			input:          "function foo()",
+			expectedName:   "foo",
+			expectedInput:  "",
+			expectedOutput: "",
+			expectedError:  nil,
+		},
+		{
+			name:           "Test case 2",
+			input:          "function bar(arg1 int, arg2 string) returns (bool)",
+			expectedName:   "bar",
+			expectedInput:  "arg1 int, arg2 string",
+			expectedOutput: "bool",
+			expectedError:  nil,
+		},
+		{
+			name:           "Test case 3",
+			input:          "function baz(arg1 int, arg2 string) returns (bool, string)",
+			expectedName:   "baz",
+			expectedInput:  "arg1 int, arg2 string",
+			expectedOutput: "bool, string",
+			expectedError:  nil,
+		},
+	}
+
+	for _, test := range tests {
+		funcName, inputArgs, outputArgs, err := ParseMethodSignature(test.input)
+		if err != test.expectedError {
+			t.Errorf("Test case %s: Expected error '%v', but got '%v'", test.name, test.expectedError, err)
+		}
+		if funcName != test.expectedName {
+			t.Errorf("Test case %s: Expected function name '%s', but got '%s'", test.name, test.expectedName, funcName)
+		}
+		if inputArgs != test.expectedInput {
+			t.Errorf("Test case %s: Expected input arguments '%s', but got '%s'", test.name, test.expectedInput, inputArgs)
+		}
+		if outputArgs != test.expectedOutput {
+			t.Errorf("Test case %s: Expected output arguments '%s', but got '%s'", test.name, test.expectedOutput, outputArgs)
+		}
+	}
```

Comment:
I don't have err variable all specified in the packages, rather declaring it on error cases. It's not possible to implement this at the moment. it will be fixed on future refactoring